### PR TITLE
Prevent disconnections caused by contract report requests

### PR DESCRIPTION
**Pull request by moonysolari**
- Previously, when hovering over a contract tile in the "My Contracts" section of the contracts menu, the user will disconnect if they press "F", which corresponds to reporting issues about the contract. 
- This PR prevents this disconnection by handling both requests related to this behavior. 
- It does not actually report anything to any server, but just returns an empty object to prevent disconnects.

### CODE REVIEW

File: components/index.ts [hunk 1]
```diff
@@ -409,6 +409,10 @@ legacyRouter.use(
     "/authentication/api/userchannel/ContractSessionsService/",
     legacyContractRouter,
 )
+legacyRouter.use(
+    "/authentication/api/userchannel/ReportingService/",
+    legacyContractRouter,
+)
```
Comment (RDIL) [inline components/index.ts:415]: Redundant: legacy router falls over to primary router if unhandled.

File: components/2016/legacyContractHandler.ts [hunk 2]
```diff
@@ -105,6 +105,17 @@ legacyContractRouter.post(
...
+legacyContractRouter.post(
+    "/ReportContract",
+    jsonMiddleware(),
+    (
+        req: RequestWithJwt<never, { contractId: string; reason: number }>,
+        res,
+    ) => {
+        res.json({})
+    },
+)
```
Comment (RDIL) [inline components/2016/legacyContractHandler.ts:117]: Redundant: legacy router falls over to primary router if unhandled.

**Review by RDIL — CHANGES_REQUESTED** [review]

**Review by RDIL — APPROVED** [review]

**Review by RDIL — APPROVED** [review]

File: components/contracts/contractRouting.ts [hunk 3]
```diff
@@ -324,6 +324,17 @@ contractRoutingRouter.post(
...
+contractRoutingRouter.post(
+    "/ReportContract",
+    jsonMiddleware(),
+    (
+        req: RequestWithJwt<never, { contractId: string; reason: number }>,
+        res,
+    ) => {
+        res.json({})
+    },
+)
```
Comment (RDIL) [inline components/contracts/contractRouting.ts:336]: This will also attach /ReportContract to /ContractsService

**Review by RDIL — CHANGES_REQUESTED** [review]

File: components/contracts/contractRouting.ts (hunk 3 above)
Comment (moonysolari) [reply components/contracts/contractRouting.ts:347]: Added a separate router for reports.

**Review by RDIL — APPROVED** [review]

File: components/contracts/contractRouting.ts (hunk 3 above)
Comment (RDIL) [reply components/contracts/contractRouting.ts:347]: Shouldn't this be removed then?

Comment (moonysolari) [reply components/contracts/contractRouting.ts]: I forgor 💀 Removed now.

**Review by RDIL — APPROVED** [review]

**Pull request merged**